app/Models/User.php

```diff
@@ -82,22 +89,98 @@ class User extends Authenticatable
+            'email' =>
+                [
+                    // Email is required when hasGlobalAdmin is true. It must be checked before nullable, this is why I
+                    // put this before the Valid::trimmed.
+                    'required_if_accepted:has_global_admin',
+                    ...Valid::trimmed([
+                        // It is a valid email address
+                        'email',
+                        // Uniqueness is only checked for insert and patch
+                        [$isInsert || $isPatch, self::getRuleUnique($original)],
+                        // If there's an email in the request, there must also be hasEmailVerified != null.
+                        new RequireNotNullRule('has_email_verified'),
+                    ],
+                        sometimes: $isPatch,
+                        nullable: true)
+                ],
+            'has_email_verified' =>
+            // If we specify hasEmailVerified in the request, we must also specify email.
+                Valid::bool([new RequireNotNullRule('email')],
+                    sometimes: $isPatch,
+                    nullable: true),
+            'password' =>
+                Valid::string([
+                    [
+                        $isInsert || $isPatch,
+                        // This is only applied to insert and patch because later on the password field gets replaced
+                        // with the hashed password.
+                        self::getPasswordRules(),
+                        // When we update the password, we must also specify expiration time, but it can be null, it
+                        // just has to be said explicitly (in the request).
+                        // new RequirePresentRule('password_expire_at')
+                        // TODO: Learn if we should use it. At the moment, the UI doesn't send this.
+                    ],
+                    // If there's a password, there must also be an email.
+                    [$isInsert || $isResource, new RequireNotNullRule('email')],
+                ],
+                    sometimes: $isPatch,
+                    nullable: true),
+            'password_expire_at' =>
+            // Password expiration is a valid datetime and if the date defined, the password must not be null.
+                Valid::datetime([new RequireNotNullRule('password')], sometimes: $isPatch, nullable: true),
+            // TODO: When password is null, password_expire_at must be null as well
+            'has_password' => Valid::bool([
+                // It must be true if the user is a global admin.
+                [$isResource || $isInsert, 'accepted_if:has_global_admin,true']
+            ]),
+            // A valid boolean
+            'has_global_admin' => Valid::bool(sometimes: $isInsert || $isPatch, nullable: true)
+        ];
+    }
+
+    public static function getInsertValidator(): array
+    {
+        return static::validationRules(false, true, false);
```

Comment:
to nie jest czytelne to te boole znaczą

wydaje mi się, że przekazanie tutaj RuleContext i rozbicie go na boole w valudation rules

albo przynajmniej z nazwami: `(isResource: false, isInsert:  true, isPatch: false);`